util/multierror/error.go

```diff
@@ -32,3 +37,14 @@ func errorFormat(errs []error) string {
 	return strings.Join(s, ", ")
 
 }
+
+type TSMultiError struct {
+	*multierror.Error
+	mu sync.Mutex
+}
+
+func (m *TSMultiError) Append(errs ...error) {
```

Comment:
Non-thread-safe version returns itself, this - is not. Why introduce such inconsistency?

Reply:
because this one is a mutating method, not a standalone function. Returning itself would be misleding, because usage is different(you don't need to pss itself as a first param).